framework/src/main/java/org/checkerframework/common/basetype/BaseTypeValidator.java

```diff
@@ -223,16 +242,20 @@ public Void visitDeclared(AnnotatedDeclaredType type, Tree tree) {
 
         final boolean skipChecks = checker.shouldSkipUses(type.getUnderlyingType().asElement());
 
-        if (!skipChecks) {
+        if (!checkTopLevelDeclaredType && !skipChecks) {
```

Comment:
Related to the "logic wrong way around" comment above: I would not expect a negation here, given the variable name.